### PR TITLE
[liberate] Added liberate formula to migrate RedHat clones

### PR DESCRIPTION
**Pull request by mackdk**
Add liberate formula from https://github.com/mmmmmmpc/liberate-formula

FIXES https://github.com/SUSE/spacewalk/issues/23012

### CODE REVIEW

File: liberate-formula/liberate/init.sls [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  file.managed:
+    - name: /etc/sysconfig/liberated
+    - contents: |
+        LIBERATED="{{ liberated }}"
+        LIBERATED_FROM="{{ osName }} {{ release }}"
+        LIBERATED_DATE="{{ liberationDate }}"
+        LIBERATED_REINSTALLED="{{ reinstallPackages }}"
+
+{% endif %} # end if file /etc/sysconfig/liberated exists
+{% endif %} # endif of rhel family
```
Comment (witekest) [inline liberate-formula/liberate/init.sls:168]: Nit: Empty lines missing here and in the following two files.

**Review by witekest — COMMENTED** [review]
Looks good at the first glance.

Why have you decided to license it under GPL? Does it reuse some GPL code? The rest of the repository is under Apache 2.

**Comment by rjmateus** [comment]
This project have a make file that can be usefull to make versions: https://github.com/SUSE/salt-formulas/blob/master/Makefile

**Comment by mackdk** [comment]
> Why have you decided to license it under GPL? Does it reuse some GPL code? The rest of the repository is under Apache 2.

I honestly just copied the license from the original repository by @mmmmmmpc. Not sure if there was a specific reason behind it, or if there are any issues in switching license now. Maybe @rjmateus knows?

**Comment by mackdk** [comment]
> This project have a make file that can be usefull to make versions: https://github.com/SUSE/salt-formulas/blob/master/Makefile

Yes, I updated it:

```makefile
liberate:: clean
	git archive --format=tar.gz --prefix=liberate-formula-${LIBERATE_VERSION}/ HEAD:liberate-formula/ >liberate-formula-${LIBERATE_VERSION}.tar.gz
```

Do we need additional changes?

**Comment by rjmateus** [comment]
yes, I need an additional change in my glasses :-)
sorry for the noise

**Comment by rjmateus** [comment]
You can change the license to be the one in this repo. I'm sure Miguel will not mind either.

**Comment by witekest** [comment]
> You can change the license to be the one in this repo. I'm sure Miguel will not mind either.

Provided that he had not reused any GPL code. @mmmmmmpc could you confirm please?

**Comment by rjmateus** [comment]
The code in this formula is not importing anything from other repositories. It just calls salt-core modules/states.

**Comment by rjmateus** [comment]
I have tested so far the changes for of the logos for:
- RHEL 8
- RHEL 9
- Centos 8
- Centos 7

It also tested the current implementation of the formula :)
I didn't look at how the highstate looks like, after migration since it should not apply to the states again.
I will look at that tomorrow and also test in Oracle linux, Alma Linux and Rocky Linux.

**Comment by rjmateus** [comment]
I have opened a PR to your branch with the changes. I think you can merge to your branch and it will appear in here. If not feel free to cherry-pick or copy the code (is not that much)

**Comment by mackdk** [comment]
I integrated the logo changes and changed the license to Apache 2.0. 

@rjmateus What about Centos 7 conflict mentioned in https://github.com/mmmmmmpc/liberate-formula/issues/1#issuecomment-1830639254 ? Is that a non issue in the end?

@witekest Do we have some automation to create the package in OBS or do I have to manually create the SR for the new `liberate-formula` package?

**Comment by rjmateus** [comment]
I tested 2 times with CentOS 7 and was unable to reproduce the error.
I also tested Oracle Linux 7,8 and 9, and Rocky 9 and they are working as expected.
I didn't teste RHEL7, neither Rocky 8 or Alma Linux 8 and 9.

From my point of view it's good to go

**Comment by mackdk** [comment]
I tested with CLM clones instead of the product channels, both with no modules and with adding multiple modules.
I did not encounter any issue with Oracle Linux and Rocky so for me it's looks ready to merge as well.

**Comment by rjmateus** [comment]
Did you tested with migration only or also with re-install?

**Comment by mackdk** [comment]
> Did you tested with migration only or also with re-install?

I reinstalled all the packages. Here's the initial part of the log for Oracle linux 9 for example:

```
Last metadata expiration check: 0:00:02 ago on Fri Jan 12 15:37:08 2024.
Dependencies resolved.
=============================================================================================================================================
 Package                         Arch    Version                       Repository                                                        Size
=============================================================================================================================================
Reinstalling:
 acl                             x86_64  2.3.1-3.el9                   susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64      71 k
 audit-libs                      x86_64  3.0.7-101.el9_0.2             susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     117 k
 authselect                      x86_64  1.2.3-7.el9                   susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     122 k
 authselect-libs                 x86_64  1.2.3-7.el9                   susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     218 k
 avahi                           x86_64  0.8-15.el9                    susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     287 k
 avahi-libs                      x86_64  0.8-15.el9                    susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64      66 k
 basesystem                      noarch  11-13.el9                     susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     7.0 k
 bash                            x86_64  5.1.8-4.el9                   susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     1.7 M
 bzip2-libs                      x86_64  1.0.8-8.el9                   susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64      40 k
 ca-certificates                 noarch  2020.2.50-94.el9              susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     350 k
 checkpolicy                     x86_64  3.3-1.el9                     susemanager:sll9-flat-no-module-default-sll-as-9-updates-x86_64  339 k
 cpio                            x86_64  2.13-16.el9                   susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     270 k
 cracklib                        x86_64  2.9.6-27.el9                  susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64      81 k
 cracklib-dicts                  x86_64  2.9.6-27.el9                  susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     3.6 M
 cronie                          x86_64  1.5.7-5.el9                   susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     114 k
 cronie-anacron                  x86_64  1.5.7-5.el9                   susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64      32 k
 crontabs                        noarch  1.11-27.20190603git.el9_0     susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64      18 k
 crypto-policies                 noarch  20220223-1.git5203b41.el9_0.1 susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64      54 k
 crypto-policies-scripts         noarch  20220223-1.git5203b41.el9_0.1 susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64      81 k
 cryptsetup-libs                 x86_64  2.4.3-4.el9                   susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     433 k
 curl                            x86_64  7.76.1-14.el9_0.4             susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     296 k
 cyrus-sasl-lib                  x86_64  2.1.27-20.el9                 susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     764 k
 dbus-broker                     x86_64  28-5.el9                      susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     171 k
 dejavu-sans-fonts               noarch  2.37-18.el9                   susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     1.3 M
 device-mapper                   x86_64  9:1.02.183-4.el9              susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     140 k
 device-mapper-event             x86_64  9:1.02.183-4.el9              susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64      34 k
 device-mapper-event-libs        x86_64  9:1.02.183-4.el9              susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64      33 k
 device-mapper-libs              x86_64  9:1.02.183-4.el9              susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     178 k
 device-mapper-persistent-data   x86_64  0.9.0-12.el9                  susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     771 k
 dhcp-client                     x86_64  12:4.4.2-15.b1.el9            susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     789 k
 dhcp-common                     noarch  12:4.4.2-15.b1.el9            susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     129 k
 diffutils                       x86_64  3.7-12.el9                    susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64     388 k
 dmidecode                       x86_64  1:3.3-7.el9                   susemanager:sll9-flat-no-module-default-sll-9-updates-x86_64      87 k
[....]
 ```

**Review by rjmateus — APPROVED** [review]
lets go with it

**Pull request merged**